messenger.go

```diff
@@ -248,7 +248,9 @@ func NewMessenger(
 		shutdownTasks: []func() error{
 			persistence.Close,
 			adapter.transport.Reset,
-			logger.Sync,
+			// Currently this often fails, seems like it's safe to ignore them
+			// https://github.com/uber-go/zap/issues/328
+			func() error { _ = logger.Sync; return nil },
```

Comment:
Missing parens on function call to Sync()

Reply:
If you call defer logger.Sync at logger instantiation time (if it is in main, for example) then it will be cleaned up on clean program exit.